test/server/media-controller.spec.js

```diff
@@ -0,0 +1,150 @@
+const chai = require( 'chai' );
+const expect = chai.expect;
+const request = require( 'request' );
+const config = require( '../../app/models/config-model' ).server;
+
+const IPfiltering = config[ 'ip filtering' ];
+
+/**
+ * Tests the request-filtering-agent to block SSRF attacks
+ * change testHTMLBody to the body of an html file that
+ * you are testing on. For the default, it says <i'm in.>
+ * and is hosted in testHTMLHost. 
+ */
+
+const testHTMLBody = 'i\'m in.\n';
+const enketoHost = 'http://localhost:8005';
+const testHTMLHost = 'http/localhost:1234';
+
+const requestURL = enketoHost + '/media/get/' + testHTMLHost;
+
+// TODO: Check if testHTMLHost is running
+
+describe( 'Media Controller', () => {
+
+    console.log( 'Testing request-filtering-agent on this request: ' + requestURL);
+    
+    describe( 'No Referer Request', () => {
+
+        if (!IPfiltering.allowPrivateIPAddress) {
+            if (IPfiltering.allowIPAddressList === '' ) {
+                it( 'allowPrivateIPAddress is false', () => {
+                    request( requestURL ,
+                        function(error, response, body){
+                            expect(body).to.be.equal(undefined);
+                        });
+                });    
+            } else if (IPfiltering.allowIPAddressList.includes( 'localhost' ) || IPfiltering.allowIPAddressList.includes( '127.0.0.1' )) {
+                it( 'allowPrivateIPAddress is false, but allowIPAddresslist contains: localhost or 127.0.0.1', () => {
+                    request( requestURL ,
+                        function(error, response, body){
+                            expect(body).to.be.equal(testHTMLBody);
+                        });
+                });
+                
+            } else if (IPfiltering.denyIPAddressList.includes( 'localhost' ) || IPfiltering.denyIPAddressList.includes( '127.0.0.1' )) {
+                it( 'allowPrivateIPAddress is false, but denyIPAddressList contains: localhost or 127.0.0.1', () => {
+                    request( requestURL ,
+                        function(error, response, body){
+                            expect(body).to.be.equal(undefined);
```

Comment:
If you set "allowIPAddressList": [ "localhost" ], this test still returns `undefined`.

I'm actually not sure if the Enketo app is running at all with these tests. Anyway, the previous comment will take care of that as well (once we run all tests we should know).